Improbable/Stdlib/Improbable.Stdlib/WorkerConnection.cs

```diff
@@ -170,69 +172,66 @@ public void StopSendingMetrics()
                 // Do nothing
             }
 
-            metricsTcs?.Dispose();
+            metricsCts?.Dispose();
+            metricsCts = null;
 
             metricsTask = null;
-            metricsTcs = null;
         }
 
         private static string GetDevelopmentPlayerIdentityToken(string host, ushort port, bool useInsecureConnection, string authToken, string playerId, string displayName)
         {
-            using (var pit = DevelopmentAuthentication.CreateDevelopmentPlayerIdentityTokenAsync(
+            using var pit = DevelopmentAuthentication.CreateDevelopmentPlayerIdentityTokenAsync(
                 host, port,
                 new PlayerIdentityTokenRequest
                 {
                     DevelopmentAuthenticationToken = authToken,
                     PlayerId = playerId,
                     DisplayName = displayName,
                     UseInsecureConnection = useInsecureConnection
-                }))
-            {
-                var value = pit.Get();
-
-                if (!value.HasValue)
-                {
-                    throw new AuthenticationException("Error received while retrieving a Player Identity Token: null result");
-                }
+                });
+            var value = pit.Get();
```

Comment:
Could use the `ToTask()` extension and `await` this and make this function async?